src/create/schema/parsers/optional.ts

```diff
@@ -57,6 +57,10 @@ export const isOptionalSchema = (
   }
 
   if (isZodType(zodSchema, 'ZodEffects')) {
+    // zod custom types
+    if (isZodType(zodSchema._def.schema, 'ZodAny'))
```

Comment:
Otherwise it would continue the recursion and call the isOptional on the last ZodAny object that already lost the transformation, so, it would always be optional.